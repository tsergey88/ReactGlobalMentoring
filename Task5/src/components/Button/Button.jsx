import React from "react";

export class Button extends React.Component {
  render() {
    const { title } = this.props;
    return (
      <a href="javascript:void(0);" className="btn">
        {title}
      </a>
    );
  }
}
