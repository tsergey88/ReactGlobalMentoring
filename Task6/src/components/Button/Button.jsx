// @flow
import React from "react";

type Props = {
  title: string
};

export const Button = ({ title, ...rest }: Props) => (
  <button className="btn" {...rest}>
    {title}
  </button>
);
