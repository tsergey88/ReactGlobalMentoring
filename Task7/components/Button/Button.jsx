// @flow
import React from "react";

import styles from "./style.scss";

type Props = {
  title: string
};

export const Button = ({ title, ...rest }: Props) => (
  <button className={styles.btn} {...rest}>
    {title}
  </button>
);
