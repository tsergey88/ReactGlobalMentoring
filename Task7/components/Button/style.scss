.btn {
  display: block;
  background-color: #d24979;
  padding: 5px 15px;
  color: white;
  font-weight: bold;
  text-decoration: none;
}
.btn:hover {
  color: gray;
}