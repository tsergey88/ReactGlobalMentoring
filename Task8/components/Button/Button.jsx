// @flow
import React from "react";
import injectSheet from "react-jss";

// import styles from "./style.scss";

type Props = {
  children: *,
  classes: {}
};

const styles = {
  btn: {
    border: "none",
    background: "none",
    cursor: "pointer",
    display: "block",
    backgroundColor: "#d24979",
    padding: "5px 15px",
    color: "white",
    fontWeight: "bold",
    textDecoration: "none",

    "&:hover": {
      color: "gray"
    }
  }
};

const Button1 = ({ children, classes, ...rest }: Props) => (
  <button className={classes.btn} {...rest}>
    {children}
  </button>
);

export const Button = injectSheet(styles)(Button1);
