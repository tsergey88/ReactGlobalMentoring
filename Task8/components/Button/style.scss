:local(button) {
  border: none;
  background: none;
  cursor: pointer;
}
:local(.btn) {
  display: block;
  background-color: #d24979;
  padding: 5px 15px;
  color: white;
  font-weight: bold;
  text-decoration: none;
}
:local(.btn:hover) {
  color: gray;
}
