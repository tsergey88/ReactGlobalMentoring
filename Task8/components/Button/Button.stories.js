import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";

import { Button } from "./Button";

storiesOf("Button", module).add("With some text", () => (
  <Button onClick={action("clicked")}>Search Button</Button>
));
