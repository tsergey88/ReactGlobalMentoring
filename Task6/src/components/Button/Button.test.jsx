import React from "react";
import { Button } from "./Button";
import { shallow } from "enzyme";

describe("Button component", () => {
  const title = "Title of button";
  const component = shallow(<Button title={title} />);

  it("should be render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should contain correct tag", () => {
    expect(component.find("button")).toHaveLength(1);
  });

  it("should contain correct title", () => {
    expect(component.text()).toEqual(title);
  });
});
