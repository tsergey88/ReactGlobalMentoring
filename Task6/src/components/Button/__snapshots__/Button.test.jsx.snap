// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should be render correctly 1`] = `
<button
  className="btn"
>
  Title of button
</button>
`;
